src/layouts/docs-view-layout/index.tsx

```diff
@@ -3,19 +3,43 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import SidebarSidecarWithToc, {
-	SidebarSidecarWithTocProps,
-} from 'layouts/sidebar-sidecar-with-toc'
+import { OutlineNavWithActive } from 'components/outline-nav/components'
+import { OutlineLinkItem } from 'components/outline-nav/types'
+import SidebarSidecarLayout, {
+	SidebarSidecarLayoutProps,
+} from 'layouts/sidebar-sidecar'
 import { DocsVersionAlertBanner } from './components'
 
 /**
  * Lightweight wrapper around SidebarSidecarWithToc which passes along some docs
  * specific props.
  */
-const DocsViewLayout = (props: SidebarSidecarWithTocProps) => {
+const DocsViewLayout = ({
+	outlineItems,
+	...layoutProps
+}: SidebarSidecarLayoutProps & { outlineItems: OutlineLinkItem[] }) => {
 	return (
-		<SidebarSidecarWithToc
-			{...props}
+		<SidebarSidecarLayout
+			{...layoutProps}
+			/**
+			 * Note: this slice() ensures that if the layout re-renders, then
+			 * OutlineNavWithActive re-renders.
+			 *
+			 * TODO: Find another way.
+			 * An alternate solution could be to lift up `useActiveSection`, and
+			 * find elements to monitor through a ref, rather that accessing the
+			 * DOM directly in `useActiveSection`.
+			 *
+			 * For context, with the current approach of accessing the DOM directly,
+			 * we must ensure that when the content re-renders (updating heading
+			 * DOM elements), the OutlineNav must also re-render (so that
+			 * `useActiveSection` runs setup again and finds the updated DOM elements,
+			 * rather than maintaining stale references).
+			 *
+			 * We took a spike at this in the following commit:
+			 * https://github.com/hashicorp/dev-portal/pull/1685/commits/70a034789006df2f14afae665962c4087f2a6ba1
```

Comment:
I paired with Bryce on this - I'm not exactly _happy_ with the `slice()` solution here... but at the same time, doing the "lift `useActiveSection`" work, while it may be a lil more robust since it gets to the root of the issue (changing `useActiveSection` so it doesn't access the DOM directly)  is kind of a significant refactor, and felt outside the scope of this PR. I've added a follow-up task at 🎟 [[maintenance] refactor uses of useActiveSection to not access the DOM directly](https://app.asana.com/0/1202097197789424/1204098852315293/f).

Reply:
Wow this is a sticky one, thanks for the explanation in the comment and link to the task 💯